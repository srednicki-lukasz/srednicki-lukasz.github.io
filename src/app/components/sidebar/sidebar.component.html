<aside class="h-100 d-flex flex-column align-items-center justify-content-between py-4 ps-3 pe-2">
  <nav>
    <ul class="d-flex flex-column m-0 p-0">
      <li>
        <a class="route-inactive" title="Home" routerLink="/" routerLinkActive="route-active" [routerLinkActiveOptions]="{ exact: true }">
          <i class="bi bi-person h5 m-0"></i>
        </a>
      </li>

      <li>
        <a class="route-inactive" title="Projects" routerLink="/projects" routerLinkActive="route-active">
          <i class="bi bi-braces h5 m-0"></i>
        </a>
      </li>
    </ul>
  </nav>

  <small class="text-center text-secondary" [innerHTML]="copyrightNotice()"></small>

  <footer>
    <ul class="d-flex flex-column m-0 p-0">
      <li>
        <a class="btn-link" target="_blank" [href]="socials.github.url" [title]="socials.github.name">
          <i class="bi bi-github h5 m-0"></i>
        </a>
      </li>

      <li>
        <a class="btn-link" target="_blank" [href]="socials.linkedin.url" [title]="socials.linkedin.name">
          <i class="bi bi-linkedin h5 m-0"></i>
        </a>
      </li>

      <li>
        <a class="btn-link" target="_blank" [href]="socials.mailto.url" [title]="socials.mailto.name">
          <i class="bi bi-envelope-fill h5 m-0"></i>
        </a>
      </li>
    </ul>
  </footer>
</aside>
