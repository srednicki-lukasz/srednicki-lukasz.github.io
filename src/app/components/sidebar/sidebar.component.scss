aside {
  > nav > ul,
  > footer > ul {
    gap: 2rem;
  }

  > small {
    writing-mode: sideways-lr;
  }
}
