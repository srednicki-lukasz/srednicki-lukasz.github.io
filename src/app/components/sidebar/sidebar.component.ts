import { ChangeDetectionStrategy, Component, computed } from '@angular/core';
import { RouterLink, RouterLinkActive } from '@angular/router';
import { authorInfo } from '../../author-info';

@Component({
  selector: 'sidebar',
  templateUrl: './sidebar.component.html',
  styleUrl: './sidebar.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [RouterLink, RouterLinkActive],
})
export class SidebarComponent {
  readonly socials = authorInfo.socials;

  copyrightNotice = computed(this.computeCopyrightNotice);

  private computeCopyrightNotice(): string {
    return `&copy; ${new Date().getFullYear()} Łukasz Średnicki<br>All rights reserved`;
  }
}
