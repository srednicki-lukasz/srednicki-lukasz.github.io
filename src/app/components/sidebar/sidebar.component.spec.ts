import { importProvidersFrom } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { SidebarComponent } from './sidebar.component';

describe('SidebarComponent', () => {
  let spectator: Spectator<SidebarComponent>;

  const createComponent = createComponentFactory({
    component: SidebarComponent,
    providers: [importProvidersFrom(RouterTestingModule)],
    detectChanges: false,
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should match snapshot', () => {
    spectator.detectChanges();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
