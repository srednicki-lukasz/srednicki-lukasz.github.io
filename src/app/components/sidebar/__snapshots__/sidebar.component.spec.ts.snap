// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should match snapshot 1`] = `
<sidebar
  copyrightNotice={[Function Function]}
  socials={[Function Object]}
>
  <aside
    class="h-100 d-flex flex-column align-items-center justify-content-between py-4 ps-3 pe-2"
  >
    <nav>
      <ul
        class="d-flex flex-column m-0 p-0"
      >
        <li>
          <a
            class="route-inactive"
            href="/"
            routerlink="/"
            routerlinkactive="route-active"
            title="Home"
          >
            <i
              class="bi bi-person h5 m-0"
            />
          </a>
        </li>
        <li>
          <a
            class="route-inactive"
            href="/projects"
            routerlink="/projects"
            routerlinkactive="route-active"
            title="Projects"
          >
            <i
              class="bi bi-braces h5 m-0"
            />
          </a>
        </li>
      </ul>
    </nav>
    <small
      class="text-center text-secondary"
    >
      © 2025 Łukasz Średnicki
      <br />
      All rights reserved
    </small>
    <footer>
      <ul
        class="d-flex flex-column m-0 p-0"
      >
        <li>
          <a
            class="btn-link"
            href="https://github.com/srednicki-lukasz"
            target="_blank"
            title="GitHub"
          >
            <i
              class="bi bi-github h5 m-0"
            />
          </a>
        </li>
        <li>
          <a
            class="btn-link"
            href="https://www.linkedin.com/in/lukasz-srednicki"
            target="_blank"
            title="LinkedIn"
          >
            <i
              class="bi bi-linkedin h5 m-0"
            />
          </a>
        </li>
        <li>
          <a
            class="btn-link"
            href="mailto:srednickilukasz.business@gmail.com"
            target="_blank"
            title="Email"
          >
            <i
              class="bi bi-envelope-fill h5 m-0"
            />
          </a>
        </li>
      </ul>
    </footer>
  </aside>
</sidebar>
`;
